docs/crocks/Either.md

```diff
@@ -0,0 +1,103 @@
+# Either
+
+`Either l r`
+
+Crock which takes a `l` and a `r`, or just a `r`, and acts accordingly:
+* taking only `r` will proceed as `Identity`
+* taking `l` and `r` will handle errors (or results defined as `Left`) with `l` and results with `r` (or results defined as `Right`)
+
+The "`Either`" crock creates disjunction with a linear data flow.
+
+```js
+// Imperative flow
+const renderPage = user => {
+  const isAuthenticated = user ? true : false
+  return isAuthenticated ? "Home Page" : "Login Page"
+}
+
+// Declarative flow
+const isAuthenticated = u => u ? Either.Right(u) : Either.Left("Login Page")
+
+const renderPage = user =>
+  isAuthenticated(user)
+  .map(u => "Home Page")
+  .value()
+```
+
+`Either` exposes these constructors and instances:
+
+| Constructor | Instance |
+|:---|:---|
+| [`Left`](#left), [`Right`](#right), [`of`](#of) | [`inspect`](#inspect), [`value`](#value), [`type`](#type), [`equals`](#equals), [`map`](#map), [`ap`](#ap), [`of`](#of), [`chain`](#chain), [`sequence`](#sequence), [`traverse`](#traverse) |
+
+## Constructors
+
+### Left
+
+`Either m => l -> m l _`
+
+### Right
+
+`Either m => r -> m _ r`
+
+### of
+
+`Either m => a -> m _ a`
+
+## Instances
+
+### inspect
+
+`() => String`
+
+### either
+
+`Either m => m l r ~> (l -> c) -> (r -> d) -> m l d`
```

Comment:
So I changed my mind on this fold. So while we cannot prevent the `d | b` from happening or check for it in JS, we should probably not teach bad habits to people. Both function, ideally should return the same type. So let's go with something like:

`Either m => m c a ~> (c -> b) -> (a -> b) -> b`

Which would be better for everybody to make sure both return `b`.
Make sense? If not please ask.